pinot-common/src/main/java/com/linkedin/pinot/common/data/DateTimeFormatUnitSpec.java

```diff
@@ -26,14 +30,66 @@
    * Time unit enum with range from MILLISECONDS to YEARS
    */
   public enum DateTimeTransformUnit {
-    YEARS,
-    MONTHS,
-    WEEKS,
-    DAYS,
-    HOURS,
-    MINUTES,
-    SECONDS,
-    MILLISECONDS;
+    YEARS {
+      @Override
+      public long convert(long millisSinceEpoch, MutableInterval reuse) {
```

Comment:
Can we have a more intuitive name than 'reuse'?

Reply:
Re-implement the method and removed the parameter